programs/redstone-sol/src/instructions/process_redstone_payload.rs

```diff
@@ -0,0 +1,108 @@
+use crate::constants::FEED_IDS_STR;
+use crate::constants::SIGNERS;
+use crate::error::RedstoneError;
+use crate::redstone;
+use crate::state::*;
+use crate::util::*;
+use anchor_lang::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Accounts)]
+#[instruction(feed_id: FeedId)]
+pub struct ProcessPayload<'info> {
+    #[account(mut)]
+    pub user: Signer<'info>,
+
+    #[account(
+        init_if_needed,
+        payer = user,
+        space = 8 + std::mem::size_of::<PriceData>(),
+        seeds = [&u256_from_slice("price".as_bytes()), &u256_from_slice(feed_id.as_slice())],
+        bump,
+        constraint = price_account.to_account_info().owner == __program_id
+    )]
+    pub price_account: Account<'info, PriceData>,
+    pub system_program: Program<'info, System>,
+}
+
+pub fn process_redstone_payload(
+    ctx: Context<ProcessPayload>,
+    feed_id: FeedId,
+    payload: Vec<u8>,
+) -> Result<()> {
+    // block_timestamp as milis
+    let block_timestamp = Clock::get()?.unix_timestamp as u64 * 1000;
+    let config = Config {
+        block_timestamp,
+        signer_count_threshold: 3,
+        signers: SIGNERS,
+        feed_ids: FEED_IDS_STR
```

Comment:
we don't want to pass all feed_ids here, just the one included as a `feed_id` parameter 

```rust
feed_ids: vec![feed_id.to_something()]
```

Reply:
0f1624e9c5c303ab3cd6e4b497cf67c55483c747

Reply:
0f1624e9c5c303ab3cd6e4b497cf67c55483c747

Reply:
I recommended to have here `feed_ids` single-value vector passed to have full compatibility with the `rust-sdk`, but ok for now ;p